packages/db/tests/translate/test-pdr.js

```diff
@@ -0,0 +1,226 @@
+'use strict';
+
+const test = require('ava');
+
+const { constructCollectionId } = require('@cumulus/message/Collections');
+const { translateApiPdrToPostgresPdr, translatePostgresPdrToApiPdr } = require('../../dist/translate/pdr');
+
+test('translatePostgresPdrToApiPdr translates postgres record to PDR record', async (t) => {
+  const fakeCollection = { name: 'abc', version: '123' };
+  const fakeCollectionPgModel = {
+    get: () => Promise.resolve(fakeCollection),
+  };
+
+  const fakeExecution = { arn: 'arn:aws:execution:1234abcd' };
+  const fakeExecutionPgModel = {
+    get: () => Promise.resolve(fakeExecution),
+  };
+
+  const fakeProvider = { name: 'ABCprov' };
+  const fakeProviderPgModel = {
+    get: () => Promise.resolve(fakeProvider),
+  };
+
+  const timestamp = new Date();
+
+  const postgresPdr = {
+    status: 'completed',
+    name: 'acbd1234.PDR',
+    collection_cumulus_id: 1,
+    provider_cumulus_id: 1,
+    execution_cumulus_id: 1,
+    progress: 100,
+    pan_sent: false,
+    pan_message: 'N/A',
+    stats: {
+      total: 2,
+      completed: 1,
+      failed: 1,
+      processing: 0,
+    },
+    address: 'http://example.com/',
+    original_url: 'http://original.example.com/',
+    duration: 10000,
+    timestamp,
+    created_at: timestamp,
+    updated_at: timestamp,
+  };
+
+  const expectedPdr = {
+    pdrName: postgresPdr.name,
+    provider: fakeProvider.name,
+    collectionId: constructCollectionId(fakeCollection.name, fakeCollection.version),
+    status: postgresPdr.status,
+    createdAt: timestamp.getTime(),
+    progress: postgresPdr.progress,
+    execution: fakeExecution.arn,
+    PANSent: postgresPdr.pan_sent,
+    PANmessage: 'N/A',
+    stats: postgresPdr.stats,
+    address: postgresPdr.address,
+    originalUrl: postgresPdr.original_url,
+    timestamp: timestamp.getTime(),
+    duration: postgresPdr.duration,
+    updatedAt: timestamp.getTime(),
+  };
+
+  const translatedPdrRecord = await translatePostgresPdrToApiPdr(
+    postgresPdr,
+    {},
+    fakeCollectionPgModel,
+    fakeProviderPgModel,
+    fakeExecutionPgModel
+  );
+
+  t.deepEqual(
+    translatedPdrRecord,
+    expectedPdr
+  );
+});
+
+test('translatePostgresPdrToApiPdr handles optional fields', async (t) => {
+  const fakeCollection = { name: 'abc', version: '123' };
+  const fakeCollectionPgModel = {
+    get: () => Promise.resolve(fakeCollection),
+  };
+
+  const fakeExecution = { arn: 'arn:aws:execution:1234abcd' };
+  const fakeExecutionPgModel = {
+    get: () => Promise.resolve(fakeExecution),
+  };
```

Comment:
Can we remove this fake model for this test? Since the point of the test is to prove that it can work when there is no execution to look up, having this mock in here makes that a little confusing

Reply:
https://github.com/nasa/cumulus/pull/2273/commits/7ed58e6a1895c91c2d21825e119ebfa9ace452bc